components/org.wso2.carbon.identity.tag.manager.core/src/main/java/org/wso2/carbon/identity/tag/manager/core/internal/impl/TagManagementServiceImpl.java

```diff
@@ -0,0 +1,271 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.tag.manager.core.internal.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.tag.manager.TagManagementService;
+import org.wso2.carbon.identity.tag.manager.core.dao.DAOFactory;
+import org.wso2.carbon.identity.tag.manager.core.dao.TagManagementDAO;
+import org.wso2.carbon.identity.tag.manager.core.internal.TagServiceDataHolder;
+import org.wso2.carbon.identity.tag.manager.exception.TagServiceClientException;
+import org.wso2.carbon.identity.tag.manager.exception.TagServiceException;
+import org.wso2.carbon.identity.tag.manager.model.ErrorMessage;
+import org.wso2.carbon.identity.tag.manager.model.Tag;
+import org.wso2.carbon.identity.tag.manager.model.TagAssociationsResult;
+import org.wso2.carbon.identity.tag.manager.model.TagListResult;
+import org.wso2.carbon.user.api.Tenant;
+import org.wso2.carbon.user.api.UserStoreException;
+
+import java.sql.SQLException;
+import java.util.List;
+
+/**
+ * Tag management service implementation class.
+ */
+public class TagManagementServiceImpl implements TagManagementService {
+
+    private static final Log LOG = LogFactory.getLog(TagManagementService.class);
+    private String tenantUuid = getTenantUuid();
+
+    public TagManagementServiceImpl() throws TagServiceException {
+
+    }
+
+    @Override
+    public String addTag(Tag tag) throws TagServiceException {
+
+        // Check if the tag name is valid.
+        if (!isTagNameValid(tag.getName())) {
+            throw new TagServiceClientException(ErrorMessage.ERROR_CODE_EMPTY_NAME.getCode(),
+                    ErrorMessage.ERROR_CODE_EMPTY_NAME.getMessage());
+        }
+        TagManagementDAO tagManagementDAO = DAOFactory.getInstance().getTagManagementDAO();
+        // Check if tag already existing.
+        if (tagManagementDAO.loadTag(tag.getName(), tag.getType(), tenantUuid) != null) {
+            throw new TagServiceClientException(ErrorMessage.ERROR_CODE_TAG_ALREADY_EXISTS.getCode(),
+                    String.format(ErrorMessage.ERROR_CODE_TAG_ALREADY_EXISTS.getMessage(), tag.getName(),
+                            tenantUuid));
+        }
+        String tagUuid = tagManagementDAO.storeTag(tag, tenantUuid);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Provided JSON request content is not in the valid format:");
```

Comment:
why we are just printing wrong format log always

Reply:
fixed the log.